### PR TITLE
Display incidents instead of reports for similar/dissimilar incidents

### LINKED ISSUE #2799: Improve Similiar and dissimilar incidents edition UX
On Incident edition page (e.g.: https://staging-aiid.netlify.app/incidents/edit/?incident_id=659)

Both sections (`Manually-selected similar and dissimilar incidents` and `Reports`) disappear and reload on each change without a "loading" or a spinner.
In addition to that, every time I click on `Yes` or `No`, all reports change under the same incident at the same time. That is the expected behaviour but it's confusing. 

This is a screen recording to visualize what I mean.

https://github.com/responsible-ai-collaborative/aiid/assets/6564809/05d603bf-7151-4e87-9bb0-47ec29e81c6a

_Originally posted by @pdcp1 in https://github.com/responsible-ai-collaborative/aiid/issues/2797#issuecomment-2130327525_

### PR DESCRIPTION
**Pull request by clari182**
- closes #2799 
- Test URL: https://pr-2815--staging-aiid.netlify.app/
- Uses incident autocomplete input to search incident
- Displays one result per incident instead of all its reports

### CODE REVIEW

**Comment by clari182** [comment]
Hey @datherton09 , can you take a look at this PR and let me know if the new design for similar/dissimilar looks ok to you? 
The test URL is https://pr-2815--staging-aiid.netlify.app/
The changes are
- It only displays incidents instead of reports when searching by incident ID
- New tooltip with information on how to use that functionality
- Some minor design changes
Let me know what you think. Thanks!!
image

**Comment by datherton09** [comment]
@clari182 Nice! Thanks :) Where is the tooltip located?

**Comment by clari182** [comment]
@datherton09 the tooltip is located right next to `Search similar incident ID`

File: site/gatsby-site/i18n/locales/en/popovers.json [hunk 1]
```diff
@@ -78,5 +78,9 @@
     "quiet": {
         "title": "Quiet",
         "text": "Quiet reports are those that will not be published in the 'Latest Reports' section of the homepage. Quiet reports are used for reports that are useful for internal data but do not need to be promoted. If you are unsure, leave this field blank."
+    },
+    "similarIncidentSearch": {
+        "title": "Search for similar incidents",
+        "text": "Search for a similar or dissimilar incident by typing an incident ID. If the resulting incident is similar click 'Yes', if it's dissimilar click 'No'. If you're not sure click 'Not sure'. Once selected, the incident will be added to the 'Manually-selected similar and dissimilar incidents' section of the form."
```
Comment (pdcp1) [inline site/gatsby-site/i18n/locales/en/popovers.json:84]: I can't find the usage of these `similarIncidentSearch` popover

File: site/gatsby-site/playwright/e2e-full/apps/incidents.spec.ts [hunk 2]
```diff
@@ -4,190 +4,292 @@ import { init } from '../../memory-mongo';
...
+
+    await page.waitForSelector('[data-testid="flowbite-toggleswitch-toggle"]');
+    await page.locator('[data-testid="flowbite-toggleswitch-toggle"]').click();
 
-        await init({
-            aiidprod: {
-                incidents: [
-                    { incident_id: 14, title: 'Incident 14', description: 'Description 14', date: '2021-01-01' },
-                ]
+    // TODO: Fix this
```
Comment (pdcp1) [inline site/gatsby-site/playwright/e2e-full/apps/incidents.spec.ts:165]: Is this `TODO:` still needed?

File: site/gatsby-site/src/components/incidents/IncidentForm.js [hunk 3]
```diff
@@ -223,45 +247,52 @@ function IncidentForm() {
...
-            header={'Manually-selected similar and dissimilar incidents'}
-            reports={editorSimilarIncidentReports}
-            loading={false}
+            incidents={editorSimilarIncidents}
+            loading={editorSimilarIncidentReportsQuery.loading}
             setFieldValue={setFieldValue}
             editId={false}
             error={false}
+            notFoundText={
+              'No similar/dissimilar incidents assigned to this incident. Use the search above to assign similar incidents.'
```
Comment (pdcp1) [inline site/gatsby-site/src/components/incidents/IncidentForm.js:287]: Translation missing

File: site/gatsby-site/src/components/incidents/IncidentForm.js [hunk 4]
```diff
@@ -223,45 +247,52 @@ function IncidentForm() {
...
+              <RelatedIncidentsArea
+                columnKey={'byId'}
+                header={similarIncidentsById.length > 0 ? 'Incidents search results' : ''}
+                incidents={similarIncidentsById}
+                loading={similarReportsByIdQuery.loading}
+                setFieldValue={setFieldValue}
+                editId={false}
+                error={false}
+                notFoundText={
+                  'No similar incidents found. Please enter an incident ID above to perform the search.'
```
Comment (pdcp1) [inline site/gatsby-site/src/components/incidents/IncidentForm.js:271]: Translation missing

File: site/gatsby-site/src/components/incidents/IncidentForm.js [hunk 5]
```diff
@@ -223,45 +247,52 @@ function IncidentForm() {
...
+              name="incidentSearch"
+              multiple={false}
+              placeHolder={t(`Search similar/dissimilar Incident Id`)}
+            />
+
+            {similarReportsByIdQuery.loading && <Spinner size={'sm'} className="mt-2 ml-1" />}
+            {similarIncidentsById.length > 0 && (
+              <RelatedIncidentsArea
+                columnKey={'byId'}
+                header={similarIncidentsById.length > 0 ? 'Incidents search results' : ''}
```
Comment (pdcp1) [inline site/gatsby-site/src/components/incidents/IncidentForm.js:264]: Translation missing

**Review by pdcp1 — CHANGES_REQUESTED** [review]
Great UI improvements! 👏 

While playing with the UI, I found an issue after clicking the "Not sure" button. A multi-state is displaying. Please take a look at the attached video for a clear reference.



https://github.com/user-attachments/assets/101a8765-42e5-415b-aeda-188f03444d10

**Review by pdcp1 — APPROVED** [review]
It looks good to me now 👍

**Comment by cesarvarela** [comment]
@clari182 please merge staging; everything should be ✅

**Pull request merged**